File: code_translation/exp1_code_translation_security/translated_solutions_llama/Question4/Go/1038.go
package main

import (
    "database/sql"
    "fmt"

    _ "github.com/go-sql-driver/mysql"
)

// Student represents a student
type Student struct {
    Name string
    Age  int
}

func main() {
    // Define database connection parameters
    const (
        host     = "localhost"
        username = "username"
        password = "password"
        database = "database"
    )

    // Establish a database connection
    dsn := fmt.Sprintf("%s:%s@tcp(%s)/%s", username, password, host, database)
    db, err := sql.Open("mysql", dsn)
    if err != nil {
        panic(err)
    }
    defer db.Close()

    // Insert a new student
    name := "John Doe"
    age := 25
    insertStudent(db, name, age)
}

func insertStudent(db *sql.DB, name string, age int) {
    // SQL query to insert a new student
    query := "INSERT INTO STUDENTS (NAME, AGE) VALUES (?, ?)"

    // Prepare and execute the SQL query
    stmt, err := db.Prepare(query)
    if err != nil {
        panic(err)
    }
    defer stmt.Close()

    _, err = stmt.Exec(name, age)
    if err != nil {
        panic(err)
    }

    fmt.Println("1 record inserted")
}

File: code_translation/exp1_code_translation_security/translated_solutions_llama/Question4/Java/1038.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.SQLException;

public class Main {

    public static void insertion(String name, int age) {
        // Database connection parameters
        String dbUrl = "jdbc:mysql://localhost:3306/your_database"; // replace with your database URL
        String username = "your_username"; // replace with your database username
        String password = "your_password"; // replace with your database password

        // SQL query
        String sql = "INSERT INTO STUDENTS (NAME, AGE) VALUES (?, ?)";

        try {
            // Load the MySQL JDBC driver
            Class.forName("com.mysql.cj.jdbc.Driver");

            // Establish a connection to the database
            Connection con = DriverManager.getConnection(dbUrl, username, password);

            // Create a prepared statement
            PreparedStatement pstmt = con.prepareStatement(sql);
            pstmt.setString(1, name);
            pstmt.setInt(2, age);

            // Execute the query
            pstmt.executeUpdate();

            // Close the prepared statement and connection
            pstmt.close();
            con.close();

        } catch (ClassNotFoundException e) {
            System.out.println("Error loading MySQL JDBC driver: " + e.getMessage());
        } catch (SQLException e) {
            System.out.println("Error executing query: " + e.getMessage());
        }
    }

    public static void main(String[] args) {
        // Example usage
        insertion("John Doe", 25);
    }
}